flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -297,17 +310,23 @@ private boolean executeStatement(String statement, ExecutionMode executionMode)
     }
 
     private void validate(Operation operation, ExecutionMode executionMode) {
-        ResultMode mode = executor.getSessionConfig(sessionId).get(EXECUTION_RESULT_MODE);
-        if (operation instanceof QueryOperation
-                && executionMode.equals(ExecutionMode.NON_INTERACTIVE)
-                && !mode.equals(TABLEAU)) {
-            throw new IllegalArgumentException(
-                    String.format(
-                            "In non-interactive mode, it only supports to use %s as value of %s when execute query. Please add 'SET %s=%s;' in the sql file.",
-                            TABLEAU,
-                            EXECUTION_RESULT_MODE.key(),
-                            EXECUTION_RESULT_MODE.key(),
-                            TABLEAU));
+        if (executionMode.equals(ExecutionMode.INITIALIZATION)) {
+            if (operation instanceof QueryOperation || operation instanceof ModifyOperation) {
```

Comment:
I suggest you list all the operations you support in init file instead of listing the ones you don't support. 
This will prevent you leaking some newly introduced operations such `StatementSetOperation`. 
AFAIK `StatementSetOperation` shouldn't be support in this init file.
And for operations like `explain`, I also don't think it's meaningful to support it in init file